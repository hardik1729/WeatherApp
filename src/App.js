import React from 'react';
import logo from './logo.svg';
import './App.css';
import Dropdown from './Dropdown'

class App extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      selected: props.selected,
      data: [],
    }

    this.selectionDidChange = this.selectionDidChange.bind(this);
  }

  selectionDidChange (newValue) {
    let data=fetch("http://open.mapquestapi.com/directions/v2/alternateroutes?key=86e3wu3G1SG1k22vRkXsygIwUE8wh6zU&from=mumbai&to=Indore")
    .then(response => {response.json()})
    this.setState({
      selected: newValue,
      data: data,
    });
  }

  render () {
    return (
      <div className="App">
        <Dropdown selectionDidChange={this.selectionDidChange}/>
        <header className="App-header">
          {
            this.state.selected ? (
              <p>
                {this.state.data}
              </p>  
            ) :
            (
              <div>
                <img src={logo} className="App-logo" alt="logo" />
                <p>
                  Weather Details to be displayed here
                </p>
              </div> 
            )
          }
        </header>
      </div>
    );
  }
}

export default App;