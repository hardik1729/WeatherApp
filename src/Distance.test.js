import React from 'react';
import TestRenderer from 'react-test-renderer';
import App from './App';

it('it renders place distances correctly', () => {
    const app = TestRenderer.create(<App />);
    let places = [
        {place:"Liverpool",distance:100},
        {place:"Bristol",distance:200},
        {place:"Manchester",distance:300}
    ];
    places.forEach(place => {
        app.root.instance.setState({
            selected: place.place,
            data: {route: {distance: place.distance}}
        });
        app.root.instance.selectionDidChange(place.place);
        expect(app.toJSON()).toMatchSnapshot();
    });
});