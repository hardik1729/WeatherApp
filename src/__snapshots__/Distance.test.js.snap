// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders place distances correctly 1`] = `
<div
  className="App"
>
  <div
    className="dropdown"
    style={
      Object {
        "background": "#61dafb",
        "position": "left",
        "width": "200px",
      }
    }
  >
    <div
      className="button"
      onClick={[Function]}
    >
      Location
    </div>
  </div>
  <header
    className="App-header"
  >
    <div>
      <p>
        Liverpool
      </p>
      <p>
        100
         miles
      </p>
    </div>
  </header>
</div>
`;

exports[`it renders place distances correctly 2`] = `
<div
  className="App"
>
  <div
    className="dropdown"
    style={
      Object {
        "background": "#61dafb",
        "position": "left",
        "width": "200px",
      }
    }
  >
    <div
      className="button"
      onClick={[Function]}
    >
      Location
    </div>
  </div>
  <header
    className="App-header"
  >
    <div>
      <p>
        Bristol
      </p>
      <p>
        200
         miles
      </p>
    </div>
  </header>
</div>
`;

exports[`it renders place distances correctly 3`] = `
<div
  className="App"
>
  <div
    className="dropdown"
    style={
      Object {
        "background": "#61dafb",
        "position": "left",
        "width": "200px",
      }
    }
  >
    <div
      className="button"
      onClick={[Function]}
    >
      Location
    </div>
  </div>
  <header
    className="App-header"
  >
    <div>
      <p>
        Manchester
      </p>
      <p>
        300
         miles
      </p>
    </div>
  </header>
</div>
`;
